From mboxrd@z Thu Jan  1 00:00:00 1970
Received: from mail-oa1-f53.google.com (mail-oa1-f53.google.com [209.85.160.53])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id EAC1421FF2E
	for <linux-wireless@vger.kernel.org>; Thu, 19 Feb 2026 00:40:24 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org; arc=none smtp.client-ip=209.85.160.53
ARC-Seal:i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1771461626; cv=none; b=cxwFcVM0u6Xcgw0A4U6M84MfM6DNAoSffrCud6dWT5R7tykn4FRMOOSjnt7/yzAadqNZ6o0oL4nm6UUdnJCfRRRo/pLuZu6XXEgSVhM6NcTNWjWKhGZujcroejlFoYnUpG6edcQoxx8NPa/23t+gHaJldaqlOBZwHmrlGf99vMM=
ARC-Message-Signature:i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1771461626; c=relaxed/simple;
	bh=W14Gh6alDKJouiozs1MRmYZeBj9IA20q4tSL2ep7v6U=;
	h=From:To:Cc:Subject:Date:Message-ID:In-Reply-To:References:
	 MIME-Version; b=t2wEcVfYVmNLpj2IUFrcSA+/JX/JRrMLoYZo65FYUfnjYtD5eGgz6El2MWTeaaOrfu+tqZ+Xu4bJ/D4m66UQTuBNakAZ7oMRPB435StOYc3BOsUnTqB8JW2EP1kHPmK4/uwapIsXtvb1wlxrohzRvB9tIx3O4mbn33viMmLbo4g=
ARC-Authentication-Results:i=1; smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org; spf=pass smtp.mailfrom=gmail.com; arc=none smtp.client-ip=209.85.160.53
Authentication-Results: smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org
Authentication-Results: smtp.subspace.kernel.org; spf=pass smtp.mailfrom=gmail.com
Received: by mail-oa1-f53.google.com with SMTP id 586e51a60fabf-40fb2789476so1415627fac.1
        for <linux-wireless@vger.kernel.org>; Wed, 18 Feb 2026 16:40:24 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1771461624; x=1772066424;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-gg:x-gm-message-state:from
         :to:cc:subject:date:message-id:reply-to;
        bh=B891xQfqJbMXoe3QxqQiLDv6Y0EESVbArn0hKUogRb4=;
        b=pMJ6k51k4xzxN44uAA7vwwWmOsuriRt3ectE2l4/W1Kpcb6Lf95i0IH+DP35VTaCCI
         0dWSoJP7//ZGqL45jvqRfKPFB3ds302xOlNPmozPaNvHquVHn3qWxz23FMMfUsLAAvdN
         DhOyOxvszJfilY8zQYpihs47/JI5pmp7asfE8gTVv9zi+um4yVFnfTar4im61kUFYOM4
         pEDiY5o1ehbqpOdtBcnqiJEw1Jo7my8TruKbAuayBpxWVz9JRUHWfmLBvTdV9J0zBwil
         gezQmkCfnA92nWaDlcBpD7wIRQiqX4kh18rAJZh36gF0prg0Q1uERzee5WbJ04hE43B0
         TbJQ==
X-Gm-Message-State: AOJu0Yx/FK/CDD+O7f/JzOKdsGxaavVS3mIYrSMwIIV0S9j+PAmrvmau
	7q1CWcokBqaCRT2nqqfSaP8XrkzbInPNJUsCEi2Obwedib8n4WFH6gst
X-Gm-Gg: AZuq6aJarjiFjdthjIRvRN9tNDYxWzfdlhPmOc2DCjjA6I1OFdya5SXEK+be0JjEJQ9
	FcfvvXAVPEhcSCD+3U8QeACUwWFUsNKOi/HaorWqICw2MSUHlV/LtZZLZ8iD+5khhKLCx8zXiZV
	X9dQfBe9gY67K1SX4eukOf8m951zc1/WyFBfo8NKmXP93tah6xy4G636bFb1l/jUOJKghBLz5Hj
	vxb++0srEyUZKPqCWLQ+TSOTXlBbcFDBnaHJOkSDDEsFMmseijNBBe70o7gLvg6iKP9PiYK5zSX
	5NJPBr5lF/7Cpdfztm0eqG8YhOT2h01d7MH/BQnNGenhZiOnPHAQM4+K4AOX8YJZgBGOKO/JoxV
	ZgR+JYXsYoqDfgpuBCQ0ZxfX5lk8gGYsiOyJ1Gs8SjI56RgdzXSPXZnV+op5R/j5ip6TBlPcFLU
	GQcCstUPmvovPH0EVxi7VvYHxZzeIbN3M035wIUyOGhNMiyuY=
X-Received: by 2002:a05:6871:825:b0:414:64dd:a2f3 with SMTP id 586e51a60fabf-4154591b202mr57947fac.29.1771461623726;
        Wed, 18 Feb 2026 16:40:23 -0800 (PST)
Received: from sean-HP-EliteBook-830-G6.lan ([207.191.35.252])
        by smtp.gmail.com with ESMTPSA id 586e51a60fabf-40eaee45d12sm23477659fac.1.2026.02.18.16.40.23
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 18 Feb 2026 16:40:23 -0800 (PST)
From: sean.wang@kernel.org
To: nbd@nbd.name,
	lorenzo.bianconi@redhat.com
Cc: linux-wireless@vger.kernel.org,
	linux-mediatek@lists.infradead.org,
	Sean Wang <sean.wang@mediatek.com>
Subject: [PATCH 07/11] wifi: mt76: mt792x: ensure MCU ready before ROM patch download
Date: Wed, 18 Feb 2026 18:40:03 -0600
Message-ID: <20260219004007.19733-7-sean.wang@kernel.org>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20260219004007.19733-1-sean.wang@kernel.org>
References: <20260219004007.19733-1-sean.wang@kernel.org>
Precedence: bulk
X-Mailing-List: linux-wireless@vger.kernel.org
List-Id: <linux-wireless.vger.kernel.org>
List-Subscribe: <mailto:linux-wireless+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-wireless+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit

From: Sean Wang <sean.wang@mediatek.com>

Restart the MCU and poll FW state to ensure correct MCU status
before downloading the ROM patch.

This is a prerequisite for enabling MT7902 PCIe and has been
validated on MT7921 and MT7925 since they share the common code path.

Signed-off-by: Sean Wang <sean.wang@mediatek.com>
---
 drivers/net/wireless/mediatek/mt76/mt792x_core.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/wireless/mediatek/mt76/mt792x_core.c b/drivers/net/wireless/mediatek/mt76/mt792x_core.c
index 2142fcc4ae27..152cfcca2f90 100644
--- a/drivers/net/wireless/mediatek/mt76/mt792x_core.c
+++ b/drivers/net/wireless/mediatek/mt76/mt792x_core.c
@@ -926,6 +926,13 @@ int mt792x_load_firmware(struct mt792x_dev *dev)
 {
 	int ret;
 
+	mt76_connac_mcu_restart(&dev->mt76);
+
+	if (!mt76_poll_msec(dev, MT_CONN_ON_MISC, MT_TOP_MISC_FW_STATE,
+			    MT_TOP_MISC2_FW_PWR_ON, 1000))
+		dev_warn(dev->mt76.dev,
+			 "MCU is not ready for firmware download\n");
+
 	ret = mt76_connac2_load_patch(&dev->mt76, mt792x_patch_name(dev));
 	if (ret)
 		return ret;
-- 
2.43.0


